OneCommunityGlobal/HighestGoodNetworkApp
# Ilya remove task modal from reports

# Description
<img width="748" alt="view-all-bug" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105448137/0ab3ae8b-661e-46c2-94fc-883d272aa9d2">

Other than converting a modal to an expanded box (see above), some bugs were noticed: 
- Total Hours were inconsistent between the minimized chart and the expanded chart
- On screens smaller than 670px the Pie Chart legend overflows and is hidden
- Console errors related to mapping arrays of data to components (no key property was provided)

# Priority: High

## Related PRS (if any):
None
…

## Main changes explained:
- Removed the modal, made the full Pie Chart expand down without covering the table under it 
- Added logic for the minimized Pie Chart to account for the tasks not being displayed if there are more than 4 total tasks. The last item on the legend will say "x other tasks" and the total hours will be consistent
- Special case: when the number of tasks is 5, all 5 will be displayed on the chart
- Added media queries to render the legend underneath the Pie Chart for screens narrower than 670px and reduced the padding for screens narrower than 435px
- fixed existing bugs related to array mapping and prop destructuring


## How to test:
1. check into current branch on the front end and "development" branch on the backend
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as admin/owner user
5. [Follow these instructions](https://www.loom.com/share/8bc672c323b64955aac4bce46c3794aa?sid=b05f5604-87cb-4ca1-80b8-49bd823893ee) to add a user account you want to test to a project and assign at least 6 tasks to that user
6. The tasks with no hours logged will not be displayed on the Pie Chart. To add hours logged, connect to **hgnproddb** using Mongo Compass, select the "tasks" collection, search for the task you need by **taskName** and add some number (not zero) to the "hoursLogged" field
  
<img width="1430" alt="task-compass" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105448137/89229872-2e1d-42fa-a4b4-c66c789f14b1">

Also feel free to use the tasks from the video below

7. Go to the Home Page -> Reports dropdown -> Reports -> People -> Search to find the user account you assigned tasks to and click on it -> test the functionality by clicking "View all"
8. Test the same feature on screen sizes 435px to 670px wide and 350-435px wide

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105448137/7258181b-dd19-45bb-b206-8c6423a51fd7



## Notes:
- Feel free to test by viewing my accounts "Ilya Manager" (30+ tasks), "Ilya Volunteer" (11 tasks) and "Ilya Newman" (5 tasks - special case)
- The console is going to have some errors not related to this PR. These can be disregarded because the code changed by this PR will not introduce these errors to the "development" branch when it gets merged
